Signal/src/ViewControllers/ContactViewController.swift

```diff
@@ -77,44 +75,38 @@ class ContactViewController: OWSViewController, ContactShareViewHelperDelegate {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
-        // Use "dark" style navigation bar in this view.
-        UIUtil.applyDefaultSystemAppearence()
+        // Use "dark" style status bar in this view,
+        // But we don't apply the entire default system UI (UIUtil.applyDefaultSystemAppearence) because
+        // doing so while pushing onto an existing nav controller who's navbars are hidden corrupts the
+        // title label/navbar items elsewhere in the stack.
+        CurrentAppContext().setStatusBarStyle(.default)
```

Comment:
To avoid corrupting the hidden navbars label color, we only mutate the status bar style, which is the only side effect  of the `applySystemStyle` we needed here anyway.

Reply:
Huh.  So does that imply that it's not always safe to use `applySignalAppearence` and `applyDefaultSystemAppearence`?